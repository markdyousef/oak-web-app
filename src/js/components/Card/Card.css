:root {
    --primary-color: #5ed892;
    --main-black-color: #131517;
    --light-black-color: #2d3438;
    --dark-gray-color: #737a80;
    --light-gray-color: #f2f2f2;
}

.container {
    cursor: pointer;
    width: 200px;
    height: 200px;
    background-color: #fff;
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
    padding: 15px;
}

.content {
    width: 100%;
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
}

.content h1 {
    font-size: 18px;
    font-weight: bold;
}
