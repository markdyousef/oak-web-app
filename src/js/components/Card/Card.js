// @flow
import React, { Component } from 'react';
import moment from 'moment';
import Dropdown from '../shared/Dropdown';
import CommentsIcon from '../../icons/comments';
import LikesIcon from '../../icons/likes';
import DotsIcon from '../../icons/dots';
import {
    Container,
    Header,
    User,
    LabelWrapper,
    Labels,
    Label,
    Paragraph,
    Main,
    Bottom,
    Time,
    Icon,
    Settings,
    H1,
    H2,
    P,
    Tooltip
} from './styles';

type DefaultProps = {
    title: null,
    content: null,
    labels: [],
    comments: [],
    likes: 0
};
type Props = {
    id: string,
    content: Object,
    cover: ?{
        urlThumb512: string
    },
    creator: {
        name: string,
        username: string,
        gravatar: ?string,
        avatar: ?{
            urlThumb64: ?string
        }
    },
    labels: ?Array<Object>,
    comments: ?Array<Object>,
    updatedAt: string,
    likes: ?Array<string>,
    removeCard: (id: string) => Promise<>,
    likeCard: (id: string) => Promise<>,
    unlikeCard: (id: string) => Promise<>,
    isLiked: bool,
    showComments: () => void,
    onShow: () => void,
    userId: string,
    updateCard: (field:Object) => void
};
type State = {
    showOptions: bool,
    likes: Array<string>,
    isLiked: bool
};

export default class Card extends Component<DefaultProps, Props, State> {
    static defaultProps:DefaultProps;
    state: State;
    props: Props;
    constructor() {
        super();
        this.state = {
            showOptions: false,
            likes: [],
            isLiked: false
        };
    }
    componentWillMount() {
        const { likes, isLiked } = this.props;
        this.setState({
            likes: (likes) || [],
            isLiked
        });
    }
    removeCard = () => {
        const { id, removeCard, updateCard } = this.props;
        removeCard(id)
            .then(() => {
                updateCard({
                    key: 'shouldUpdate',
                    value: true
                });
            })
            .catch((err) => {
                console.log(err);
            });
    }
    handleLike = () => {
        const { likeCard, unlikeCard, id, userId } = this.props;
        const { likes, isLiked } = this.state;
        const newLikes = [...likes];

        if (isLiked) {
            unlikeCard(id)
                .then(() => {
                    this.setState({
                        isLiked: false,
                        likes: newLikes.filter(label => label !== userId)
                    })
                })
                .catch((err) => console.log(err));
        } else {
            likeCard(id)
                .then(() => {
                    newLikes.push(userId);
                    this.setState({
                        isLiked: true,
                        likes: newLikes
                    })
                })
                .catch((err) => { throw err; });
        }
    }
    formatContent = (content: Object) => {
        const { blocks } = content;
        return (
            <div>
                {blocks.map((block) => {
                    switch (block.type) {
                    case 'header-one':
                        return <H1 key={block.key}>{block.text}</H1>;
                    case 'header-two':
                        return <H2 key={block.key}>{block.text}</H2>;
                    }
                })}
                <Paragraph>
                {blocks.map((block) => {
                    switch (block.type) {
                    case 'unstyled':
                        return <P key={block.key}>{block.text}</P>;
                    }
                })}
                </Paragraph>
            </div>
        );
    }
    // renderHeader = () => {
    //     const { creator: { name, username } } = this.props;
    //     return (
    //         <Header>
    //         </Header>
    //     );
    // }
    renderContent = () => {
        const { updatedAt, content, onShow, cover, labels } = this.props;
        let coverImg;
        if (cover && cover.urlThumb512) coverImg = cover.urlThumb512;
        return (
            <Main onClick={onShow}>
                {coverImg && <img alt="card" src={coverImg} />}
                {content && this.formatContent(content)}
                { /* <button onClick={onShow}>Read more</button> */ }
                    <Labels>
                        {labels && labels.map((label, index) => {
                            const right = 16 + (index * 12);
                            return (
                            <LabelWrapper key={label.id}>
                                <Label
                                    style={{ background: label.color, right }}
                                />
                                <span>{label.name}</span>
                            </LabelWrapper>
                            );
                        }).reverse()}
                    </Labels>
                <Time>
                    {moment(updatedAt).fromNow()}
                </Time>
            </Main>
        );
    }
    renderBottom = () => {
        const { comments, showComments, creator: { name, avatar, gravatar } } = this.props;
        const { likes, isLiked, showOptions } = this.state;
        let picture;
        // prefer avatar over gravatar
        if (gravatar) picture = gravatar;
        if (avatar) picture = avatar.urlThumb64;
        return (
            <Bottom>
                <div>
                    <Icon onClick={showComments}>
                        <span data-title="Comment" data-title-pos="bottom">
                            <CommentsIcon />
                        </span>
                        <p>{(comments) ? comments.length : 0}</p>
                    </Icon>
                    <Icon onClick={this.handleLike}>
                        <span data-title="Like" data-title-pos="bottom">
                            <LikesIcon isLiked={isLiked} />
                        </span>
                        <p>{likes.length}</p>
                    </Icon>
                    <Icon onClick={() => this.setState({ showOptions: !showOptions })}>
                        <span data-title="More" data-title-pos="bottom">
                            <DotsIcon />
                        </span>
                        {showOptions &&
                            <Settings>
                                <Dropdown
                                    arrowPos="left"
                                    onClose={() => this.setState({ showOptions: false })}
                                >
                                    <Icon onClick={this.removeCard}>Delete post</Icon>
                                </Dropdown>
                            </Settings>
                        }
                    </Icon>
                </div>
                    <span data-title={name} data-title-pos="bottom">
                    <User>
                            <img src={picture} alt="avatar" />
                    </User>
                </span>
            </Bottom>
        );
    }
    render() {
        return (
            <Container>
                {/* {this.renderHeader()} */}
                {this.renderContent()}
                {this.renderBottom()}
            </Container>
        );
    }
}
