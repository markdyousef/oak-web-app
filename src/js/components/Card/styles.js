import styled from 'styled-components';
import colors from '../../styles/colors';

const button = () => {
    return `
        border: none;
        background-color: transparent;
        cursor: pointer;
        padding: 0
    `;
};

export const Container = styled.div`
    width: 320px;
    background-color: ${colors.white};
    border: 1px solid ${colors.lightGrey};
    border-radius: 3px;
    padding: 16px 16px 20px;
    margin: 8px;
    position: relative;
    &:hover {
        background: #fafafa;
    }
    &:active {
        background: #f7f7f7;
    }
`;

export const Header = styled.div`
    width: 100%;
    display: flex;
`;

export const User = styled.div`
    display: flex;
    align-items: center;
    & img {
        height: 25px;
        width: 25px;
        border-radius: 999em;
    }
`;

export const LabelWrapper = styled.div`
    padding: 6px 8px;
    border: 1px solid #e5e5e5;
    display: inline-block;
    border-radius: 3px;
    margin-right: 8px;
    margin-bottom: 4px;
    & span {
        color: rgba(19, 21, 23, 0.8);
        font-size: 14px;

    }
`;

export const Labels = styled.div`
    position: relative;
    display: block;
    padding-top: 8px;
`;

export const Label = styled.div`
    width: 10px;
    height: 10px;
    border-radius: 3px;
    line-height: 1.54;
    margin-right: 4px;
    display: inline-block;
`;

export const Main = styled.div`
    margin-bottom: 8px;
    text-align: left;
    cursor: pointer;
    & h1 {
        font-size: 24px;
        color: rgba(19, 21, 23, 0.9);
        letter-spacing: -0.12px;
        line-height: 1.15;
        font-weight: bold;
        overflow: hidden;
        display: -webkit-box;
        -webkit-line-clamp: 2;
        -webkit-box-orient: vertical;
    }
    & h2 {
        font-size: 20px;
        color: rgba(19, 21, 23, 0.5);
        letter-spacing: -0.08px;
        padding-top: 8px;
        max-width: 100%;
        text-overflow: ellipsis;
        white-space: nowrap;
        overflow: hidden;
    }
    & p {
        padding-top: 12px;
        font-size: 16px;
        color: rgba(19, 21, 23, 0.8);
        line-height: 1.38;
        overflow: hidden;
        display: -webkit-box;
        -webkit-line-clamp: 3;
        -webkit-box-orient: vertical;
    }
    & img {
        max-width: 288px;
        width: 100%;
        margin-bottom: 16px;
        ${''/* height: 100px; */}
    }
    & button {
        margin-top: 16px;
        ${button()};
        color: ${colors.grey}
    }
`;

export const Paragraph = styled.div`
    max-height: 4.1em;
    overflow: hidden;
`;

export const Bottom = styled.div`
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: flex-end;
    & div {
        display: flex;
        align-items: center;
    }
    & span {
        display: flex;
        align-items: center;
    }
    & [data-title] {
        position: relative;
        &:after {
            position: absolute;
            font-size: 14px;
            border-radius: .4rem;
            content: attr(data-title);
            padding: 6px 10px;
            color: rgba(255,255,255,.8);
            background-color: #000;
            box-shadow: 0 0 14px rgba(0, 0, 0, 0.1);
            color: $wenk-font-color;
            line-height: 1.25rem;
            text-align: left;
            z-index: 1;
            pointer-events: none;
            display: block;
            opacity: 0;
            visibility: hidden;
            left: 50%;
            transform: translate(-50%, 10px);
            white-space: pre;
            width: auto;
        }
        &:before {
            content: '';
            height: 10px;
            width: 10px;
            transform: rotate(45deg);
            position: absolute;
            left: 0;
            right: 0;
            margin-left: auto;
            margin-right: auto;
            border-radius: 3px;
            top: 31px;
            background: #000;
            opacity: 0;
            visibility: hidden;
        }
        &:hover {
            overflow: visible;
            &:after {
                display: block;
                opacity: 1;
                visibility: visible;
                transform: translate(-50%, 10px);
            }
            &:before {
                opacity: 1;
                visibility: visible;
            }

            &[data-title-pos="bottom"] {
                &:after {
                    bottom: auto;
                    top: 100%;
                    left: 50%;
                    transform: translate(-50%, 10px);
                }
                &:hover {
                    &:after {
                        transform: translate(-50%, 10px);
                    }
                }
            }
        }
`;

export const Time = styled.span`
    font-size: 12px;
    color: ${colors.grey};
    padding-top: 8px;
    display: block;
`;

export const Icon = styled.div`
    height: 18px;
    display: flex;
    align-items: center;
    margin-right: 16px;
    cursor: pointer;
    position: relative;
    & p {
        color: ${colors.black};
        font-size: 14px;
        margin-left: 4px;
    }
    & svg {
        height: 100%;
        height: 18px;
    }
    & [data-title] {
        position: relative;
        &:after {
            position: absolute;
            font-size: 14px;
            border-radius: .4rem;
            content: attr(data-title);
            padding: 6px 10px;
            color: rgba(255,255,255,.8);
            background-color: #131517;
            box-shadow: 0 0 14px rgba(0, 0, 0, 0.1);
            color: $wenk-font-color;
            line-height: 1.25rem;
            text-align: left;
            z-index: 1;
            pointer-events: none;
            display: block;
            opacity: 0;
            visibility: hidden;
            left: 50%;
            transform: translate(-50%, 10px);
            white-space: pre;
            width: auto;
        }
        &:before {
            content: '';
            height: 10px;
            width: 10px;
            transform: rotate(45deg);
            position: absolute;
            left: 0;
            right: 0;
            margin-left: auto;
            margin-right: auto;
            border-radius: 3px;
            top: 24px;
            background: #131517;
            opacity: 0;
            visibility: hidden;
        }
        &:hover {
            overflow: visible;
            &:after {
                display: block;
                opacity: 1;
                visibility: visible;
                transform: translate(-50%, 10px);
            }
            &:before {
                opacity: 1;
                visibility: visible;
            }

            &[data-title-pos="bottom"] {
                &:after {
                    bottom: auto;
                    top: 100%;
                    left: 50%;
                    transform: translate(-50%, 10px);
                }
                &:hover {
                    &:after {
                        transform: translate(-50%, 10px);
                    }
                }
            }
        }
    }
`;

export const Settings = styled.div`
    position: absolute;
    width: 160px;
    bottom: -72px;
    left: -22px;
    z-index: 99;
    display: block !important;
    & div div div {
        position: relative;
        z-index: 1;
        margin-right: 0;
        width: 100%;
        height: initial;
    }
    & div div div:last-child {
        margin-bottom: 0;
    }
    & div div div:hover {
        color: #fff;
    }
    & div div div:hover:before {
        content: '';
        background: #F76664;
        border-radius: 3px;
        padding: 4px 8px;
        color: #fff;
        position: absolute;
        margin: -8px;
        width: calc(100% + 16px);
        height: 32px;
        z-index: -1;
    }
`;

export const H1 = styled.h1`
    width: 100%;
`;

export const H2 = styled.h2`
    width: 100%;
`;

export const P = styled.p`
    width: 100%;
`;
