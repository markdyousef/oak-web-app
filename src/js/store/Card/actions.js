// @flow
import { EditorState, convertFromRaw } from 'draft-js';
import { decorator } from 'zen-editor';
import * as types from '../constants/ActionTypes';
import type { Action } from './reducers';

type Field = {
    key: string,
    value: any
}

export const updateCard = (field: Field): Action => {
    return ({
        type: types.CARD_STATE,
        data: {
            field
        }
    });
};

export const clearCard = () => (
    {
        type: types.CLEAR_CARD
    }
);

type Card = {
    id: string,
    name?: string,

}
export const setCard = (card: Card) => (
    {
        type: types.SET_CARD,
        data: { card }
    }
);

export const setCardContent = (content: string) => {
    // create editorstate based on content (string)
    let editorState;
    try {
        editorState = JSON.parse(content);
    } catch (e) {
        editorState = null;
    }
    if (editorState !== null && typeof editorState === 'object') {
        const state = convertFromRaw(editorState);
        editorState = EditorState.createWithContent(state, decorator)
        return {
            type: types.SET_CARD_CONTENT,
            data: { content: editorState }
        }
    }
    return {
        type: types.SET_CARD_CONTENT,
        data: { message: 'wrong format!'}
    }
}

export const addCardImage = (image:Object) => (
    {
        type: types.SET_CARD_IMAGE,
        data: { image }
    }
)
