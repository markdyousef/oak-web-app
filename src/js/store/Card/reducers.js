// @flow
import { fromJS, Map, List } from 'immutable';
import * as types from '../constants/ActionTypes';
import { EditorState, convertFromRaw } from 'draft-js';
import { decorator } from 'zen-editor';

// type State = {
//     isLoading: bool,
//     shouldUpdate: bool,
//     cardId: ?string,
//     collectionId: ?string,
//     message: ?Map<string, string>
// };

type State = Map<string, any>

export type Action = {
    type?: string,
    data?: Object,
    result?: Object,
    queryId?: string
};

const initialState:State = Map({
    queryId: null,
    isLoading: false,
    shouldUpdate: false,
    cardId: null,
    collectionId: null,
    message: null,
    editorState: EditorState.createEmpty(decorator),
    images: List([]),
    name: '',
    readOnly: false,
    menu: null,
    isSaved: false
});

const convertToEditor = (content: string):EditorState => {

    let state;
    try {
        state = JSON.parse(content);
    } catch (e) {
        state = null;
    }
    if (state !== null && typeof state === 'object') {
        const contentState = convertFromRaw(state);
        return EditorState.createWithContent(contentState, decorator);

    }
    return EditorState.createEmpty(decorator);
}

export default (state: State = initialState, action: Action): State => {
    switch (action.type) {
    // case types.UPDATE_CARD:
    //     return state.set('shouldUpdate', action.shouldUpdate);
    case types.CARD_STATE:
        if (action.data && action.data.field) {
            const { data: { field } } = action;
            return state.set(field.key, fromJS(field.value));
        }
        return state;
    case types.CLEAR_CARD:
        return state
            .set('cardId', null)
            .set('collectionId', null)
            .set('editorState', EditorState.createEmpty(decorator))
    case types.SET_CARD:
        if (action.data && action.data.card) {
            const { data: { card } } = action;
            return state
                .set('isLoading', false)
                .set('cardId', card.id)
                .set('message', null)
                .set('images', List([]))
        }
        return state;
    case types.SET_CARD_CONTENT:
        if (action.data && action.data.content) {
            const { data: { content } } = action;
            return state.mergeIn(['editorState'], content);
        }
        return state;
    case types.SET_CARD_IMAGE:
        if (action.data && action.data.image) {
            const { data: { image } } = action;
            return state.set('images', state.get('images').push(fromJS(image)));
        }
    case 'APOLLO_QUERY_RESULT':
        if (action.operationName === 'getCard' && action.result) {
            const { result: { data }, queryId } = action;
            if (data.seed && data.seed.content && queryId) {
                const editorState = convertToEditor(data.seed.content);
                console.log(queryId);
                return state.mergeIn(['editorState'], editorState).set('queryId', queryId);
            }
            return state;
        }
        return state;
    case 'APOLLO_QUERY_RESULT_CLIENT':
        if (action.queryId === state.get('queryId') && action.result) {
            const { result: { data }, queryId } = action;
            if (data.seed && data.seed.content && queryId) {
                console.log(data.seed);
                const editorState = convertToEditor(data.seed.content);
                return state.mergeIn(['editorState'], editorState).set('queryId', queryId);
            }
        }
        return state;
    default:
        return state;
    }
};
