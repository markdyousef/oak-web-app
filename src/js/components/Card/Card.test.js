import React from 'react';
import renderer from 'react-test-renderer';
import Card from './Card';

it('renders correctly', () => {
    const props = {
        creatorId: '1',
        updatedAt: '',
        onShow: () => {},
        onRemove: () => {},
        onLike: () => {},
        showComments: () => {},
        creator: { avatar: {} }
    };
    const component = renderer.create(
        <Card {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
