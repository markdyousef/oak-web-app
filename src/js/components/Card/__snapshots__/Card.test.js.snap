exports[`test renders correctly 1`] = `
<div
  className="iQUjGL">
  <div
    className="lmwPmP">
    <div />
    <div
      className="dwVQkf">
      <img
        alt="avatar"
        src={undefined} />
      <div>
        <h4 />
        <h5>
          @
        </h5>
      </div>
    </div>
  </div>
  <div
    className="ewliVH">
    <img
      alt="card"
      src={null} />
    <button
      onClick={[Function]}>
      Read more
    </button>
  </div>
  <div
    className="fdBgn">
    <div>
      <div
        className="eEtODH"
        onClick={[Function]}>
        <svg
          height="16"
          viewBox="0 0 16.6 16"
          width="16.6">
          <g>
            <g
              id="XMLID_1_">
              <g>
                <path
                  d="M8.05,1.2c-3.82,0-6.93,2.85-6.93,6.37s3.11,6.38,6.93,6.38c0.86,0,1.71-0.15,2.51-0.43c0.17-0.08,0.38-0.06,0.53,0.08 c0.03,0.02,0.94,0.78,2.82,1.03c-0.76-1.15-0.63-2.65-0.63-2.73c0.01-0.11,0.06-0.21,0.13-0.29c1.04-1.19,1.57-2.57,1.57-4.04 C14.98,4.06,11.87,1.2,8.05,1.2z M8.05,14.95c-4.37,0-7.93-3.31-7.93-7.38c0-4.06,3.56-7.37,7.93-7.37s7.93,3.3,7.93,7.37 c0,1.66-0.61,3.28-1.71,4.57c-0.01,0.45,0.05,1.77,0.92,2.4c0.23,0.17,0.33,0.46,0.24,0.73c-0.09,0.26-0.33,0.43-0.61,0.43 c0,0-0.01,0-0.02,0c-2.26-0.06-3.6-0.81-4.13-1.17C9.83,14.81,8.95,14.95,8.05,14.95z" />
              </g>
            </g>
          </g>
        </svg>
        <span>
          0
        </span>
      </div>
      <div
        className="eEtODH"
        onClick={[Function]}>
        <svg
          height="16"
          viewBox="0 0 18 16"
          width="18">
          <path
            d="M17.74,5.32c0-2.72-2.21-4.94-4.94-4.94c-1.55,0-2.98,0.72-3.9,1.92C7.98,1.1,6.55,0.38,5,0.38c-2.73,0-4.95,2.22-4.94,4.89 C0.03,5.5-0.41,10.8,8.64,15.43c0.01,0.01,0.02,0.01,0.03,0.02c0.04,0.01,0.07,0.03,0.1,0.03c0.04,0.01,0.09,0.02,0.13,0.02 c0.04,0,0.09-0.01,0.13-0.02c0.03,0,0.06-0.02,0.1-0.03c0.01-0.01,0.02-0.01,0.03-0.02C18.21,10.8,17.76,5.5,17.74,5.32z M8.9,14.43 c-8.18-4.26-7.85-9.01-7.85-9.11c0-2.17,1.77-3.94,3.95-3.94c1.44,0,2.77,0.79,3.46,2.05c0.05,0.08,0.11,0.15,0.2,0.2 c0.24,0.14,0.55,0.05,0.68-0.2c0.69-1.26,2.02-2.05,3.46-2.05c2.17,0,3.94,1.77,3.95,4C16.75,5.42,17.14,10.14,8.9,14.43z" />
        </svg>
        <span>
          0
        </span>
      </div>
      <div
        className="eEtODH"
        onClick={[Function]}>
        <svg
          height="16"
          viewBox="0 0 16 16"
          width="16">
          <g>
            <path
              d="M2.891,10.151C1.705,10.151,0.74,9.185,0.74,8c0-1.186,0.965-2.151,2.151-2.151S5.042,6.814,5.042,8 C5.042,9.185,4.077,10.151,2.891,10.151z M2.891,6.749C2.201,6.749,1.64,7.31,1.64,8c0,0.69,0.561,1.251,1.251,1.251 S4.142,8.689,4.142,8C4.142,7.31,3.581,6.749,2.891,6.749z" />
          </g>
          <g>
            <path
              d="M8,10.151c-1.186,0-2.151-0.965-2.151-2.151c0-1.186,0.965-2.151,2.151-2.151S10.151,6.814,10.151,8 C10.151,9.186,9.186,10.151,8,10.151z M8,6.749C7.31,6.749,6.749,7.31,6.749,8C6.749,8.69,7.31,9.251,8,9.251S9.251,8.69,9.251,8 C9.251,7.31,8.69,6.749,8,6.749z" />
          </g>
          <g>
            <path
              d="M13.109,10.151c-1.186,0-2.151-0.965-2.151-2.151c0-1.186,0.965-2.151,2.151-2.151S15.26,6.814,15.26,8 C15.26,9.185,14.295,10.151,13.109,10.151z M13.109,6.749c-0.69,0-1.251,0.561-1.251,1.251c0,0.69,0.561,1.251,1.251,1.251 S14.36,8.689,14.36,8C14.36,7.31,13.799,6.749,13.109,6.749z" />
          </g>
        </svg>
      </div>
    </div>
    <span
      className="cKgiew">
      Invalid date
    </span>
  </div>
</div>
`;
